### PR TITLE
Changed parameters, loss functions in variational autoencoder example

### PR DESCRIPTION
**Pull request by michielree**
Fixed some (what I thought be) mistakes in the variational autoencoder example

1. KL div between latent distribution and its prior is now summed rather than averaged over the latent variables
2. Same holds for the likelihood of the output variables
3. The std of the noise distribution is now incorporated into KLD part of the loss function, since it affects the effective std's of the latent distriution. Here, my rationale was that the std of the noise might be a hyperparameter you want to optimize, and that to be able to do that the loss functions have to be comparable.
4. Earlier KL div missed scaling of the log_std by 2

Note that points 1 & 2 are now in line with the implementation referred to by Durk Kingma: https://github.com/y0ast/Variational-Autoencoder/blob/master/VAE.py

Additional changes:
- Increased the batch size (helps with stability), plus Kingma+Welling argue that they can get away with sampling the latent distribution only once during training (as we do here) with a batch size >= 100
- Changed to plotting range to accommodate the new range of the latent dist.

### CODE REVIEW

**Comment by michielree** [comment]
ok somehow i missed https://github.com/fchollet/keras/pull/3220, apologies, closing
